docs/root/updating_envoy_dependency.md

```diff
@@ -0,0 +1,463 @@
+   all lines that are unique to Nighthawk are marked with comment `# unique`.
+
+#### Example commands
+
+```bash
+merge_from_envoy "tools/code_format/config.yaml"
+```
+
+### Step 11
+If [requirements.txt](requirements.txt) has not been updated in the last 30 days (based on comment at top
+   of file), check for major dependency updates.
+
+#### Example commands
+
+```bash
+head -1 requirements.txt
+```
+
+- If less than 30 days ago, skip to next step.
+- If more than 30 days ago, do the rest of this step.
+
+##### In brief
+
+```bash
+virtualenv pip_update_env
+source pip_update_env/bin/activate
+pip install -r requirements.txt
+pip list --outdated
+```
+
+```bash
+vi requirements.txt
+```
+
+```bash
+deactivate
+rm -rf pip_update_env
+```
+
+##### In detail
+
+1. Create and activate a virtual env:
+   ```bash
+   virtualenv pip_update_env
+   source pip_update_env/bin/activate
+   ```
+   NOTE: if `pip_update_env/bin/activate` appears to not exist, try setting
+   `export DEB_PYTHON_INSTALL_LAYOUT='deb'` and recreating the virtualenv.
+
+1. Install dependencies:
+
+   ```bash
+   pip install -r requirements.txt
+   ```
+
+1. Check for outdated dependencies:
+
+   ```bash
+   pip list --outdated
+   ```
+   This will likely show both outdated dependencies based on requirements.txt and other outdated
+   dependencies you may have in addition, such as to `pip` itself. Here, we are only interested in
+   cross-referencing the ones that appear with the ones in `requirements.txt`.
+
+1. If you find any dependency updates, you can either try updating the dependency in
+   `requirements.txt` yourself or create an issue for the change and assign it to one of the
+   nighthawk maintainers. If you do it yourself, note that all versions must be pinned to conform
+   with `rules_python`.
+
+   If there are not any dependency updates, please update the timestamp at the top of the file.
+
+1. When done, clean up the virtual env:
+
+   ```bash
+   deactivate
+   rm -rf pip_update_env
+   ```
+
+### Step 12
+
+Run:
+
+```bash
+ci/do_ci.sh test
+```
+
+Sometimes the dependency update comes with changes
+   that break our build. Include any changes required to Nighthawk to fix that
+   in the same PR.
+
+If the tests fail, try rerunning a few times.
+
+If there are build failures or persistent test failures, you will need to fix them at this point.
+
+See [Troubleshooting](#troubleshooting) for tips.
+
+### Step 13
+
+If the PR ends up modifying any C++ files, execute:
+
+```bash
+rm -rf tools/pyformat/
+ci/do_ci.sh fix_format
+```
+
+to reformat the files and avoid a CI format check failure.
+
+
+### Step 14
+
+Execute:
+
+```bash
+tools/update_cli_readme_documentation.sh --mode fix
+```
+
+to regenerate the
+   portion of our documentation that captures the CLI help output. This will
+   prevent a CI failure in case any flags changed in the PR or upstream.
+
+### Step 15
+
+Create a PR with a title like `Update Envoy to 9753819 (Jan 24th 2021)`,
```

Comment:
Let's link the following PR as an example of how to write the PR description: 

https://github.com/envoyproxy/nighthawk/pull/758

Reply:
Done.